### PR TITLE
Implement deleteAllById() in SimpleSpannerRepository

### LINKED ISSUE #567: Implement `deleteAllById()` in `SimpleSpannerRepository`
Spring Data 2.5 introduced a new method in `CrudRepository`: [`deleteAllById()`](https://github.com/spring-projects/spring-data-commons/pull/476). Since we support both Spring Boot 2.4 and 2.5, this means we'll have to omit `@Overrides` annotation to maintain ability to compile against 2.4.

### PR DESCRIPTION
**Pull request by zhumin8**
Implement deleteAllById() in SimpleSpannerRepository and add relevant tests.
Resolves #567

### CODE REVIEW

File: spring-cloud-gcp-data-spanner/src/test/java/com/google/cloud/spring/data/spanner/SimpleSpannerRepositoryTests.java [hunk 1]
```diff
@@ -1,42 +0,0 @@
-/*
```
Comment (zhumin8) [inline spring-cloud-gcp-data-spanner/src/test/java/com/google/cloud/spring/data/spanner/SimpleSpannerRepositoryTests.java:1]: There is a SpannerRepositoryImplTests in the repo that already has some test on `SimpleSpannerRepository`, so deleted this and added unit test there instead.

File: spring-cloud-gcp-data-spanner/src/test/java/com/google/cloud/spring/data/spanner/repository/support/SpannerRepositoryImplTests.java [hunk 2]
```diff
@@ -264,6 +265,20 @@ public void deleteByIdTest() {
...
+
+		when(this.entityProcessor.convertToKey(Key.of("key1"))).thenReturn(Key.of("key1"));
+		when(this.entityProcessor.convertToKey(Key.of("key2"))).thenReturn(Key.of("key2"));
+
+		new SimpleSpannerRepository<Object, Key>(this.template, Object.class)
+				.deleteAllById(unconvertedKey);
+		ArgumentCaptor<KeySet> argumentCaptor = ArgumentCaptor.forClass(KeySet.class);
+		verify(this.template).delete(eq(Object.class), argumentCaptor.capture());
+		assertThat(argumentCaptor.getValue().getKeys()).containsExactlyInAnyOrder(Key.of("key2"), Key.of("key1"));
+	}
```
Comment (zhumin8) [inline spring-cloud-gcp-data-spanner/src/test/java/com/google/cloud/spring/data/spanner/repository/support/SpannerRepositoryImplTests.java:280]: this `delete()` method itself already tested below.  
https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/main/spring-cloud-gcp-data-spanner/src/test/java/com/google/cloud/spring/data/spanner/core/SpannerTemplateTests.java#L685-L698

File: spring-cloud-gcp-data-spanner/src/test/java/com/google/cloud/spring/data/spanner/repository/support/SpannerRepositoryImplTests.java [hunk 3]
```diff
@@ -264,6 +265,20 @@ public void deleteByIdTest() {
 		verify(this.template, times(1)).delete(Object.class, A_KEY);
 	}
 
+	@Test
+	public void deleteAllByIdTest() {
+		List<Key> unconvertedKey = Arrays.asList(Key.of("key1"), Key.of("key2"));
+
+		when(this.entityProcessor.convertToKey(Key.of("key1"))).thenReturn(Key.of("key1"));
```
Comment (suztomo) [inline spring-cloud-gcp-data-spanner/src/test/java/com/google/cloud/spring/data/spanner/repository/support/SpannerRepositoryImplTests.java:272]: (Probably not directly related to this PR) This `convertToKey` behavior seems strange. It's takes a `Key` and returning `Key`. I see it's not converting anything. Do you know why?

Comment (zhumin8) [reply spring-cloud-gcp-data-spanner/src/test/java/com/google/cloud/spring/data/spanner/repository/support/SpannerRepositoryImplTests.java:272]: This `convertToKey` method itself can take in a `Key`, a single key component, or an array of key components to convert. You are right it's not doing anything in this case.  I tried to keep it simple and consistent with other tests as it's not the point of the test.

Comment (suztomo) [reply spring-cloud-gcp-data-spanner/src/test/java/com/google/cloud/spring/data/spanner/repository/support/SpannerRepositoryImplTests.java:272]: Got it. Thanks.

**Review by elefeint — CHANGES_REQUESTED** [review]
Looks great, just a couple of minor things.

File: spring-cloud-gcp-data-spanner/src/test/java/com/google/cloud/spring/data/spanner/repository/support/SpannerRepositoryImplTests.java (hunk 3 above)
Comment (zhumin8) [reply spring-cloud-gcp-data-spanner/src/test/java/com/google/cloud/spring/data/spanner/repository/support/SpannerRepositoryImplTests.java:272]: hmm, it does feel less odd this way. Let me make the change.

**Review by elefeint — APPROVED** [review]

**Pull request merged**